lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -8,24 +10,41 @@ class PokemonOverviewPage extends StatelessWidget {
     Key? key,
   }) : super(key: key);
 
-  final List<Pokemon> pokemons;
+  final Async<List<Pokemon>> pokemons;
 
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
       home: Scaffold(
-        appBar: AppBar(title: const Text('Pokedex')),
-        body: GridView.builder(
-          gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(crossAxisCount: 2),
-          itemCount: pokemons.length,
-          itemBuilder: (_, index) {
-            final pokemon = pokemons[index];
-            return PokemonCard(
-              pokemon: pokemon,
+        appBar: AppBar(title: const Text(appbarTitle)),
+        body: Builder(
```

Comment:
why wrap this in a Builder? seems no need for this u can maximize using the build method's provided context 🙂